freqtrade/enums/marginmode.py

```diff
@@ -0,0 +1,10 @@
+from enum import Enum
+
+
+class MarginMode(Enum):
+    """
+        Enum to distinguish between
+        one-way mode or hedge mode in Futures (Cross and Isolated) or Margin Trading
+    """
+    ONE_WAY = "one-way"
+    HEDGE = "hedge"
```

Comment:
**Conversation from the [closed PR](https://github.com/freqtrade/freqtrade/pull/5379#discussion_r700739993)**

@xmatthias xmatthias 15 days ago •

i assume this is the binance futures "hedge mode" ?

To be honest, i think if we don't start to keep things simple (focus on a few selected modes supported by multiple exchanges) - this will never have a chance to be merged.

Personally, i'd have focused on cross-margin on binance, "regular short" (probably still margin) on kraken - and futures on FTX (they seem to all work pretty similarily).
These seem pretty straightforward at a first glance (what binance futures clearly does not).

Additional modes can always be added at a later point - but adding too much in one go will make everything take forever - and we'll also want to do real trading tests of a few 100 trades for each mode/exchange before merging ... so having too many exchanges WILL become problematic for sure.
@arunavo4 arunavo4 12 days ago

@xmatthias These modes are only supported for Binance so for other exchanges its just simple margin trading which will not have any modes so if we want to test it we can just test it for binance the check is already there for that.
Member
@xmatthias xmatthias 12 days ago

My point is - we should focus on the simple things to get the whole short/leverage stuff working.
If we start supporting the "special binance modes" (whichever ones they are) - it'll become a very long endavor (please consider that we'll have to live-test each and every one of these different modes, for several trades (not just 2-3)) - so each additional mode will probably add at least 2-3 weeks of testing time alone (not talking about development time).
Member Author
@samgermain samgermain 11 days ago

@arunavo4 If you want to make it easier to get your code for hedge mode back, I would finish up the regular liquidation formulas with their tests, then after that's done, remove the hedge code in a single commit. Then when you want to add it back, you could just revert that commit
@arunavo4 arunavo4 10 days ago

Yea lets do that keep that code to add later
Member Author
@samgermain samgermain 10 days ago

    Yea lets do that keep that code to add later

Just make a good commit message that you can find later

Reply:
I have added Tests for this now